aws-s3/spring-cloud-starter-stream-source-s3/src/main/resources/META-INF/spring-configuration-metadata-whitelist.properties

```diff
@@ -0,0 +1,4 @@
+configuration-properties.classes=org.springframework.cloud.stream.app.s3.source.AmazonS3SourceProperties, \
+  org.springframework.cloud.stream.app.file.FileConsumerProperties, \
+  org.springframework.cloud.stream.app.trigger.TriggerPropertiesMaxMessagesDefaultUnlimited
```

Comment:
There is a large back story to all this.  `@EnableConfigurationProperties` is just a convenience to register the `@ConfigurationProperty` class in the context vs. using an `@Bean` style approach.  This is all about whitelisting of properties.  This is based off the property metadata generated by the boot annotation post processor, ends up as spring-configuration-metadata.json in the jar.  ATM the key issue is that for a standard boot app, there are literally hundreds of properties that are available in the `classpath:*` of those metadata files and we need a way to say - hey for this app - these are the really important ones you should be configuring.  The dataflow server is resonsible for reading this file and implementing the whitelist functionality.